### PR TITLE
improve(suggested-fees): Filter underlying simulation reasons

### PR DESCRIPTION
**Pull request by pxrl**

### CODE REVIEW

File: api/suggested-fees.ts [hunk 1]
```diff
@@ -199,16 +199,22 @@ const handler = async (
...
+        computedOriginChainId,
+        destinationChainId,
+        recipientAddress,
+        tokenPrice,
+        message,
+        relayerAddress
+      );
+    } catch {
+      throw new InputError("Failed to estimate relayer fees");
+    }
```
Comment (james-a-morris) [inline api/suggested-fees.ts:217]: We should place this in `_utils.ts` where that function is instantiated. 

We'd just need to change on line ~532 of `_utils.ts` 

```
    // Resolve and transform the error
    const e = _e as Error;
    // We want to mask this error as an Input error.
    throw new InputError(e?.message);
```
to
```
    throw new InputError("Failed to estimate relayer fees.");
```

**Review by james-a-morris — APPROVED** [review]

File: api/_utils.ts [hunk 2]
```diff
@@ -729,6 +727,14 @@ export function applyMapFilter<InputType, MapType>(
   }, []);
 }
 
+export function resolveEthersError(err: unknown): string {
+  return sdk.typeguards.isEthersError(err)
+    ? `${err.reason}: ${err.code}`
+    : sdk.typeguards.isError(err)
+    ? err.message
+    : "unknown error";
```
Comment (pxrl) [inline api/_utils.ts:735]: I think the code construction here is OK, but this auto-formatting is ++annoying. I'd prefer it to be:
```suggestion
  return sdk.typeguards.isEthersError(err)
    ? `${err.reason}: ${err.code}`
    : sdk.typeguards.isError(err)
      ? err.message
      : "unknown error";
```

Comment (pxrl) [reply api/_utils.ts]: Have manually overridden it via a prettier directive.

File: api/suggested-fees.ts (hunk 1 above)
Comment (pxrl) [reply api/suggested-fees.ts]: Great - that was actually clobbering the ethers error in the catch statement that I had, so it definitely makes sense to put it there.

**Pull request merged**